flask/templating.py

```diff
@@ -101,7 +101,20 @@ def list_templates(self):
 
 
 def _render(template, context, app):
-    """Renders the template and fires the signal"""
+    """Renders the template and fires signals.
+
+    It is possible to override render template in the before_render_template signal.
+    It can be done only if exactly one receiver and it return not None result."""
+
+    brt_resp = before_render_template.send(app, template=template, context=context)
+    
+    overrides = [rv for _, rv in brt_resp if rv is not None]
```

Comment:
One way to avoid all kinds of trouble would be moving the `_render` util into the `Flask` class. That would allow one to keep using the `flask.render_template` function (which is incredibly useful, since you have *one* place to import it from) but still customize its behavior on the application level.

I'd keep the signal nonetheless to modify the context but ignore the return value.